2/answers.txt

```diff
@@ -0,0 +1,231 @@
+with multiple solutions for the same problem, branching these allows you to pick and choose between them and merge the ones that are useful, and merge changes with the master branch.
+
+To organise them I would simply create a branch for each change you wish to make, roughly speaking, so if you were to make a change to the code to solve a particular bug or functionality requirement then you would title that branch accordingly.
+
+Allowing you to compare and merge those changes individually and check for issues that may arise or errors that are made before combining them into the master branch.
+
+Q7: Why is it important to frequently commit and push your changes to a remote repository (eg your forked repo on GitHub)?
+
+Frequent commit and push will safeguard your work against loss of data if the original is somehow compromised, as well as ensuring that with smaller consistent changes, anything that causes
+a problem elsewhere or breaks something is easily identifiable, committing a lot of work at once increases the chance of a conflict with code that is submitted and will make it harder to
+troubleshoot an error should one arise.
+
+Q8: Why are VM snapshots important?
+
+VM Snapshots when done regularly will allow you to restore a VM to a state much closer to that which it was before an error or failure occurs, saving lots of time and effort in lost work
+Ultimately saving the state of the machine like this is similar to regularly saving your work, if it goes down and you have to restore it, you will have lost less if you snapshot regularly
+restoring it to a state before it experienced a problem.
+
+Q9: When and why would you choose to use incremental backups vs full backups? 
+
+Incremental backups are especially useful when backing up large amounts of data or data that changes infrequently, by having the backup only transfer that which has changed since the
+previous backup, you can save a lot of network traffic, data transfer costs.
+
+Q10: What port is SSH usually exposed on? What are the security implications of having this port exposed to the internet? Is it worthwhile changing the port?
+
+SSH is exposed on port 22 by default, the security implications are that an attacker would know where to look for the port making a connection atttempt, changing the port would likely make
+the default operation of a lot of programs function incorrectly without additional configuration, such as any SSH connection would need to be told explicitly which port is listening for the SSH commands.
+
+I would guess a determined attacker would scan all ports on a location for the port that responds to an SSH request if thats what they were looking for and perhaps the obfuscation of the SSH port would Ultimately
+prove pointless because of this? Ultimately you might end up with less risk because bots set to attempt port 22 would miss it but ultimately if its configured correctly such an attack would be futile.
```

Comment:
It does provide some protection against bots, via obfuscation.
You are able to use any port up to 65535, if you choose 41603 for example, it would take a sophisticated bot more time and effort to scan through all 65535 ports to find one that returns a response that looks like SSH. Most brute force SSH bots aren't going to be this sophisticated.

If you use a SSH key with a secure cypher, the chances of a brute-force attack succeeding are practically 0.

What you can do is reduce the likelihood of DDOS attacks causing high load against your server using some of the following techniques:

1. Have a firewall at the edge of your network that kills off bad actors before they reach your server.

2. IP address obfuscation via reverse proxy helps if using a web server. Your server IP address will never be revealed to the public internet, only the proxy's. The server is still accessible to the internet but someone would have to guess the IP address and would never know who the server belongs to.

3. Have port 22 only accessible to internal networks (requiring a vpn to connect). But this comes with it's own problems. how do your manage the proxy instance remotely? is the vpn more resilient against brute force attacks than ssh?

4. Use brute force detecting software like fail2ban. If you get multiple port 22 hits from the same IP in a short period of time that fail to authenticate, ban the IP for a period of time.